CMakeLists.txt

```diff
@@ -237,6 +237,18 @@ if(BUILD_TESTING)
     )
     FetchContent_MakeAvailable(googletest)
 
+		# NOTE: gtest uninit some variables, gcc >= 1.11.3 may cause error on compile. 
+		# remove this func and func call, once ninja deps gtest-1.12.0 or above  
+		function(maybe_add_compile_flags)
```

Comment:
Thank you for the latest change. As an additional nit, I would suggest naming this `maybe_add_gtest_compile_flags` for clarity :-)

Node that I don't have write access to the repository, invoking @jhasse here for his feedback.

Reply:
I'm wondering why using a function at all? Couldn't this code be inlined below?

Reply:
cause i wanna make inside part looks more clear

Reply:
hm ... I would rather keep the added lines at a minimum. Can you inline it and also remove the ` message`?

Reply:
sure